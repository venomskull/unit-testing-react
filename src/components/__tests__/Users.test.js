import {cleanup, render, screen} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Users from '../Users';


// test('should render Users component', () => {
//     render(<Users />);
//     const userElement = screen.getByTestId('user-1');
//     expect(userElement).toBeInTheDocument();
//     // expect(userElement).toHaveTextContent('hello');
//     expect(userElement).toHaveTextContent('Jayahari');
// })

afterEach(() => {
    cleanup();
})

test('should render illegal user', () => {
    const user = {id: 1, name: 'Jayahari', age: 5}
    render(<Users user={user} />)
    const userElement = screen.getByTestId('user-1');
    expect(userElement).toBeInTheDocument();
    expect(userElement).toHaveTextContent('Jayahari1');
})

test('should render legal user', () => {
    const user = { id: 2, name: 'Deepan', age: 37 }
    render(<Users user={user} />)
    const userElement = screen.getByTestId('user-2');
    expect(userElement).toBeInTheDocument();
    expect(userElement).toHaveTextContent('Deepan');
    expect(userElement).toContainHTML('green');
})

// snapshot testing

test('matches snapshot', () => {
    const user = {id: 1, name: 'Jayahari', age: 5}
    const tree = renderer.create(<Users user={user} />).toJSON();
    expect(tree).toMatchSnapshot();
})