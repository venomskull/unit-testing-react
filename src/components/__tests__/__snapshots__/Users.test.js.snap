// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="user-1"
>
  <h1
    style={
      Object {
        "color": "red",
      }
    }
  >
    Jayahari
  </h1>
</div>
`;
